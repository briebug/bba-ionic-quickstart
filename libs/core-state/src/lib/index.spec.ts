import { mockLesson, mockCourse, mockAuthor, mockUser } from '@bba/testing';
import {
  getFullCourses,
  getFullAuthors,
  getFullUsers,
  getFullUser,
  getNavigatedCourse,
  getNavigatedLesson,
} from '.';

describe('Index Common Selector', () => {
  let lessons, courses, authors, users;

  beforeEach(() => {
    lessons = [
      { ...mockLesson, id: '0' },
      { ...mockLesson, id: '1' },
      { ...mockLesson, id: '2' },
    ];
    courses = [
      { ...mockCourse, id: '0' },
      { ...mockCourse, id: '1' },
      { ...mockCourse, id: '2' },
    ];
    authors = [
      { ...mockAuthor, id: '0' },
      { ...mockAuthor, id: '1' },
      { ...mockAuthor, id: '2' },
    ];
    users = [
      { ...mockUser, id: '0' },
      { ...mockUser, id: '1' },
      { ...mockUser, id: '2' },
    ];
  });

  it('should select the related authors and lessons on Courses', () => {
    expect(
      getFullCourses.projector(authors, courses, lessons)
    ).toMatchSnapshot();
  });

  it('should select the related courses on Authors', () => {
    expect(getFullAuthors.projector(authors, courses)).toMatchSnapshot();
  });

  it('should select the related courses and lessons on Users', () => {
    expect(getFullUsers.projector(courses, lessons, users)).toMatchSnapshot();
  });

  describe('should select the related courses and lessons on selected user', () => {
    it('on selected user', () => {
      expect(
        getFullUser.projector(courses, lessons, users[0])
      ).toMatchSnapshot();
    });

    it('on no selected user', () => {
      expect(getFullUser.projector(courses, lessons, null)).toMatchSnapshot();
    });
  });

  it('should select the navigated course', () => {
    courses = [
      { ...mockCourse, id: '0' },
      { ...mockCourse, id: '0' },
      { ...mockCourse, id: '0' },
    ];

    expect(getNavigatedCourse.projector(courses, '0')).toMatchSnapshot();
  });

  it('should select the navigated lesson', () => {
    expect(getNavigatedLesson.projector(lessons, 0)).toMatchSnapshot();
  });
});
