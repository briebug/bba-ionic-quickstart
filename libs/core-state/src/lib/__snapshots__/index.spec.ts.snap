// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Common Selector should select the navigated course 1`] = `
Object {
  "authorId": "0",
  "description": "mockCourse constant for testing",
  "id": "0",
  "thumbnail": "",
  "title": "mockCourse",
}
`;

exports[`Index Common Selector should select the navigated lesson 1`] = `
Object {
  "courseId": "0",
  "description": "mockLesson constant for testing",
  "id": "0",
  "title": "mockLesson",
  "videoUri": "",
}
`;

exports[`Index Common Selector should select the related authors and lessons on Courses 1`] = `
Array [
  Object {
    "author": Object {
      "description": "mockAuthor constant for testing",
      "firstName": "Mock",
      "id": "0",
      "lastName": "Author",
      "profilePic": "",
      "title": "mockAuthor",
    },
    "authorId": "0",
    "description": "mockCourse constant for testing",
    "id": "0",
    "lessons": Array [
      Object {
        "courseId": "0",
        "description": "mockLesson constant for testing",
        "id": "0",
        "title": "mockLesson",
        "videoUri": "",
      },
      Object {
        "courseId": "0",
        "description": "mockLesson constant for testing",
        "id": "1",
        "title": "mockLesson",
        "videoUri": "",
      },
      Object {
        "courseId": "0",
        "description": "mockLesson constant for testing",
        "id": "2",
        "title": "mockLesson",
        "videoUri": "",
      },
    ],
    "thumbnail": "",
    "title": "mockCourse",
  },
  Object {
    "author": Object {
      "description": "mockAuthor constant for testing",
      "firstName": "Mock",
      "id": "0",
      "lastName": "Author",
      "profilePic": "",
      "title": "mockAuthor",
    },
    "authorId": "0",
    "description": "mockCourse constant for testing",
    "id": "1",
    "lessons": Array [],
    "thumbnail": "",
    "title": "mockCourse",
  },
  Object {
    "author": Object {
      "description": "mockAuthor constant for testing",
      "firstName": "Mock",
      "id": "0",
      "lastName": "Author",
      "profilePic": "",
      "title": "mockAuthor",
    },
    "authorId": "0",
    "description": "mockCourse constant for testing",
    "id": "2",
    "lessons": Array [],
    "thumbnail": "",
    "title": "mockCourse",
  },
]
`;

exports[`Index Common Selector should select the related courses and lessons on Users 1`] = `
Array [
  Object {
    "coursesCompleted": 1,
    "currentCourse": Object {
      "authorId": "0",
      "description": "mockCourse constant for testing",
      "id": "0",
      "thumbnail": "",
      "title": "mockCourse",
    },
    "currentLesson": Object {
      "courseId": "0",
      "description": "mockLesson constant for testing",
      "id": "1",
      "title": "mockLesson",
      "videoUri": "",
    },
    "currentLessonId": "1",
    "description": "mockUser constant for testing",
    "email": "mockUser@mock.com",
    "firstName": "Mock",
    "id": "0",
    "lastName": "User",
    "lessonsCompleted": 1,
    "password": "12345",
    "profilePic": "",
    "title": "mockUser",
  },
  Object {
    "coursesCompleted": 1,
    "currentCourse": Object {
      "authorId": "0",
      "description": "mockCourse constant for testing",
      "id": "0",
      "thumbnail": "",
      "title": "mockCourse",
    },
    "currentLesson": Object {
      "courseId": "0",
      "description": "mockLesson constant for testing",
      "id": "1",
      "title": "mockLesson",
      "videoUri": "",
    },
    "currentLessonId": "1",
    "description": "mockUser constant for testing",
    "email": "mockUser@mock.com",
    "firstName": "Mock",
    "id": "1",
    "lastName": "User",
    "lessonsCompleted": 1,
    "password": "12345",
    "profilePic": "",
    "title": "mockUser",
  },
  Object {
    "coursesCompleted": 1,
    "currentCourse": Object {
      "authorId": "0",
      "description": "mockCourse constant for testing",
      "id": "0",
      "thumbnail": "",
      "title": "mockCourse",
    },
    "currentLesson": Object {
      "courseId": "0",
      "description": "mockLesson constant for testing",
      "id": "1",
      "title": "mockLesson",
      "videoUri": "",
    },
    "currentLessonId": "1",
    "description": "mockUser constant for testing",
    "email": "mockUser@mock.com",
    "firstName": "Mock",
    "id": "2",
    "lastName": "User",
    "lessonsCompleted": 1,
    "password": "12345",
    "profilePic": "",
    "title": "mockUser",
  },
]
`;

exports[`Index Common Selector should select the related courses and lessons on selected user on no selected user 1`] = `undefined`;

exports[`Index Common Selector should select the related courses and lessons on selected user on selected user 1`] = `
Object {
  "coursesCompleted": 1,
  "currentCourse": Object {
    "authorId": "0",
    "description": "mockCourse constant for testing",
    "id": "0",
    "thumbnail": "",
    "title": "mockCourse",
  },
  "currentLesson": Object {
    "courseId": "0",
    "description": "mockLesson constant for testing",
    "id": "1",
    "title": "mockLesson",
    "videoUri": "",
  },
  "currentLessonId": "1",
  "description": "mockUser constant for testing",
  "email": "mockUser@mock.com",
  "firstName": "Mock",
  "id": "0",
  "lastName": "User",
  "lessonsCompleted": 1,
  "password": "12345",
  "profilePic": "",
  "title": "mockUser",
}
`;

exports[`Index Common Selector should select the related courses on Authors 1`] = `
Array [
  Object {
    "courses": Array [
      Object {
        "authorId": "0",
        "description": "mockCourse constant for testing",
        "id": "0",
        "thumbnail": "",
        "title": "mockCourse",
      },
      Object {
        "authorId": "0",
        "description": "mockCourse constant for testing",
        "id": "1",
        "thumbnail": "",
        "title": "mockCourse",
      },
      Object {
        "authorId": "0",
        "description": "mockCourse constant for testing",
        "id": "2",
        "thumbnail": "",
        "title": "mockCourse",
      },
    ],
    "description": "mockAuthor constant for testing",
    "firstName": "Mock",
    "id": "0",
    "lastName": "Author",
    "profilePic": "",
    "title": "mockAuthor",
  },
  Object {
    "courses": Array [],
    "description": "mockAuthor constant for testing",
    "firstName": "Mock",
    "id": "1",
    "lastName": "Author",
    "profilePic": "",
    "title": "mockAuthor",
  },
  Object {
    "courses": Array [],
    "description": "mockAuthor constant for testing",
    "firstName": "Mock",
    "id": "2",
    "lastName": "Author",
    "profilePic": "",
    "title": "mockAuthor",
  },
]
`;
